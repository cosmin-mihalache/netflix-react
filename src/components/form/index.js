import React from 'react';
import {
  Container,
  Base,
  Title,
  Submit,
  Text,
  SmallText,
  Input,
  Link,
  Error,
} from './styles/form';

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Input = function FormInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Link = function FormLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.SmallText = function FormSmallText({ children, ...restProps }) {
  return <SmallText {...restProps}>{children}</SmallText>;
};
