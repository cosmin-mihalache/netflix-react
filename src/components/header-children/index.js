import React from 'react';
import { Container, Title, SubTitle } from './styles/header-children';

export default function HeaderChild({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

HeaderChild.Title = function HeaderChildTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

HeaderChild.SubTitle = function HeaderChildSubTitle({
  children,
  ...restProps
}) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};
