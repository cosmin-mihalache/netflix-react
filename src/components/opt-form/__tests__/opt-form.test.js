import React from 'react';
import { render } from '@testing-library/react';
import { OptForm } from '../../../components';

describe('OptForm', () => {
  it('renders <OptForm /> component with populated data', () => {
    const {container, getByText, getByPlaceholderText } = render(
      <OptForm>
        <OptForm.Text>
          Ready to watch? Enter your email to create or restart your membership.
        </OptForm.Text>
        <OptForm.Input placeholder="Email Address" />
        <OptForm.Button>GET STARTED</OptForm.Button>
      </OptForm>
    );

    expect(getByText('Ready to watch? Enter your email to create or restart your membership.')).toBeTruthy();
    expect(getByPlaceholderText('Email Address')).toBeTruthy();
    expect(getByText('GET STARTED')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
