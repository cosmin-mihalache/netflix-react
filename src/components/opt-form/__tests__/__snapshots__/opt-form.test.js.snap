// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptForm renders <OptForm /> component with populated data 1`] = `
<div
  class="opt-form__Container-jboqzi-0 eIFwJw"
>
  <p
    class="opt-form__Text-jboqzi-3 DWCvg"
  >
    Ready to watch? Enter your email to create or restart your membership.
  </p>
  <input
    class="opt-form__Input-jboqzi-1 jyePyv"
    placeholder="Email Address"
  />
  <button
    class="opt-form__Button-jboqzi-2 fkTjMc"
  >
    GET STARTED
    <img
      alt="GET STARTED"
      src="./images/icons/chevron-right.png"
    />
  </button>
</div>
`;
